### PR TITLE
Fixed round error on mingw

### PR DESCRIPTION
**Pull request by nobu**

### CODE REVIEW

File: core/float/round_spec.rb [hunk 1]
```diff
@@ -10,11 +10,9 @@
     0.0.round.should == 0
   end
 
-  platform_is_not "x64-mingw32" do
-    it "returns the nearest Integer for Float near the limit" do
-      0.49999999999999994.round.should == 0
-      -0.49999999999999994.round.should == 0
-    end
+  it "returns the nearest Integer for Float near the limit" do
```
Comment (eregon) [inline core/float/round_spec.rb]: This likely needs a version guard.

Comment (nobu) [inline core/float/round_spec.rb]: This is a bug in mingw runtime, and we should report it to the upstream, I think.
Once it gets fixed there, will we need a guard for mingw version?

Comment (nobu) [inline core/float/round_spec.rb]: BTW, I'm uncertain where is the most upstream of mingw :(

Comment (eregon) [inline core/float/round_spec.rb]: Is it https://sourceforge.net/p/mingw/bugs/ maybe?

Comment (nobu) [inline core/float/round_spec.rb]: Thank you, posted https://sourceforge.net/p/mingw/bugs/2325/ now.

Comment (eregon) [inline core/float/round_spec.rb]: Thanks! There was a reply on that bug.

**Comment by mjago** [comment]
I did some investigating on this:

Firstly just to be sure, under IEEE-754 double-precision 0.49999999999999994 IS representable - the following link shows the breakdown:

http://www.binaryconvert.com/result_double.html?decimal=048046052057057057057057057057057057057057057057057057052

Looking at mingw-w64 round.c it looks like no provision is taken for a result `larger’ than a double which looks like it will result in a double-rounding error:

https://github.com/Alexpux/mingw-w64/blob/b028d1b994ab601843a518a2e4ad0d74a05e9569/mingw-w64-crt/math/round.c

Here is the gcc version of round() in real.c - care is taken to ensure information is not lost during the truncation stage as commented here:

https://github.com/gcc-mirror/gcc/blob/master/gcc/real.c#L31

According to the following article GCC also used to incorrectly calculate round - but this was fixed in 2013 :

http://www.exploringbinary.com/real-c-rounding-is-perfect-gcc-now-converts-correctly/

Presumably if Mingw want to make claims of providing the Gnu Compiler Collection on Windows then they must ensure their algorithms are updated to match?

**Comment by eregon** [comment]
@nobu Is this fixed (or worked around) in ruby 2.4 or trunk?
If so, we can add a version guard here and merge it.

**Comment by eregon** [comment]
@nobu Ping :)
Was this fixed in MRI itself?
If so please commit the spec changes directly in ruby/ruby (should be easier).

**Pull request closed without merging**